### PR TITLE
CLOUDSTACK-9128: Testcase to verify physical_size attribute of snapshot_store_ref table 

### PR DESCRIPTION
**Pull request by pritisarap12**
Verify if physical_size attribute of snapshot_store_ref table stores actual physical size of the snapshot

### CODE REVIEW

**Comment by pritisarap12** [comment]
Testcase result:

Check Snapshots size in database ... === TestName: test_02_snapshot_size_check | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 337.826s

OK

File: test/integration/testpaths/testpath_snapshot_limits.py [hunk 1]
```diff
@@ -358,3 +362,129 @@ def test_01_storage_snapshots_limits(self):
...
+    @attr(tags=["advanced", "basic"], required_hardware="true")
+    def test_02_snapshot_size_check(self):
+        """ Check Snapshots size in database
+            1. Create file on ROOT disk of deployed VM.
+            2. Create Snapshot of ROOT disk.
+            3. Check if physiacal_size parameter of snapshot_store_ref table
+               has physical size of snapshot
+        """
+        if self.hypervisor.lower() not in ["xenserver", "vmware"]:
+            self.skipTest("Test not to be run on %s" % self.hypervisor)
```
Comment (remibergsma) [inline test/integration/testpaths/testpath_snapshot_limits.py]: Why can't this run on xenserver?

File: test/integration/testpaths/testpath_snapshot_limits.py [hunk 2]
```diff
@@ -358,3 +362,129 @@ def test_01_storage_snapshots_limits(self):
...
+        storage_qry_result_old = qryresult_before_snapshot[0]
+        secondary_storage_old = storage_qry_result_old[2]
+
+        createChecksum(
+            self.testdata,
+            self.vm,
+            root_volume,
+            "rootdiskdevice")
+
+        time.sleep(30)
```
Comment (remibergsma) [inline test/integration/testpaths/testpath_snapshot_limits.py]: Can you please explain why we wait 30s here?

File: test/integration/testpaths/testpath_snapshot_limits.py (hunk 1 above)
Comment (pritisarap12) [inline test/integration/testpaths/testpath_snapshot_limits.py]: Here we are running the test for Xenserver and VMWare  i.e. skip the test if the hypervisor is not XS or VMWare .
So the test run on Xenserver.

File: test/integration/testpaths/testpath_snapshot_limits.py (hunk 2 above)
Comment (pritisarap12) [inline test/integration/testpaths/testpath_snapshot_limits.py]: createChecksum function logins the VM and performs some operation on VM disk which takes some time, so wait for 30s is added there.

**Comment by pvr9711** [comment]
@remibergsma can you merge these changes?  thanks, Raja

**Comment by sanju1010** [comment]
LGTM !!

Test Results:
===========
Check Snapshots size in database ... === TestName: test_02_snapshot_size_check | Status : SUCCESS === 
ok  
---------------------------------------------------------------------- 
Ran 1 tests in 385.302s  
OK

**Comment by remibergsma** [comment]
@pvr9711 I guess you know what we need before merge, don't you? ;-)

**Comment by nitt10prashant** [comment]
LGTM
based on code review

**Comment by shwetaag** [comment]
Did code review. Code looks  good to me

**Comment by cristofolini** [comment]
@pritisarap12 Could you please rebase this against the current master?

**Comment by pritisarap12** [comment]
Rebased with current master.

**Pull request merged**